dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/SearchServiceBean.java

```diff
@@ -218,6 +219,7 @@ public SolrQueryResponse search(DataverseRequest dataverseRequest, List<Datavers
 
         solrQuery = addHighlightFields(solrQuery, solrFieldsToHightlightOnMap);
 
+        solrQuery.setHighlightRequireFieldMatch(true);
```

Comment:
While it might make sense for the `FILE_EXTENSION` field, are you sure we want that for other fields as well and it will still work for other fields as before? I'm not Solr expert, so just wondering if it makes sense for the whole query, and if so, than why wasn't it here before?

Reply:
I'm not a Solr expert either :) But this change is actually more of a tuning to #105 .
When testing #105 I've decided that it wasn't really necessary to add this option, but with this new `File Extension` field it doesn't look good when, for example, a user searches for `zip` files and in results has it highlighted on `file name`, `file extension` and `file type`.

It should work fine with all the fields - it only modifies which fields are gonna be highlighted and doesn't change search results.